fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/cursors/SizeStatisticsGroupingCursorTest.java

```diff
@@ -0,0 +1,479 @@
+
+    @Test
+    public void calcSize() throws Exception {
+        populateStore(1_000, 1_000_000L);
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            long exactStoreSize = getTotalSize(getStoreSizeResult(recordStore));
+            long exactRecordsSize = getTotalSize(getRecordsSizeResult(recordStore));
+            long exactIndexSize = getTotalSize(getIndexSizeResult(recordStore, "MySimpleRecord$num_value_3_indexed"));
+            long exactSubspaceSize = getTotalSize(getSubspaceSizeResult(recordStore, recordStore.getSubspace()));
+
+            Assertions.assertThat(exactStoreSize).isGreaterThan(0);
+            Assertions.assertThat(exactRecordsSize).isGreaterThan(0);
+            Assertions.assertThat(exactIndexSize).isGreaterThan(0);
+            Assertions.assertThat(exactStoreSize).isEqualTo(exactSubspaceSize);
+
+            commit(context); // commit as that logs the store timer stats
+        }
+    }
+
+    @Test
+    public void calcSizeWithContinuation() throws Exception {
+        populateStore(1_000, 1_000_000L);
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            ScanProperties scanProperties = new ScanProperties(ExecuteProperties.newBuilder().setScannedBytesLimit(10000).build());
+            // Scan a little of the data
+            RecordCursorResult<SizeStatisticsGroupedResults> continuationResult = getStoreSizeResult(recordStore, scanProperties, null, 0);
+            Assertions.assertThat(continuationResult.hasNext()).isFalse(); // should return early due to BYTE LIMIT, no value returned
+            Assertions.assertThat(continuationResult.getNoNextReason()).isEqualTo(RecordCursor.NoNextReason.BYTE_LIMIT_REACHED);
+            Assertions.assertThat(continuationResult.getContinuation()).isNotNull();
+            // scan the rest of the data
+            continuationResult = getStoreSizeResult(recordStore, ScanProperties.FORWARD_SCAN, continuationResult.getContinuation().toBytes(), 0);
+            Assertions.assertThat(continuationResult.hasNext()).isTrue();
+            Assertions.assertThat(continuationResult.getContinuation().isEnd()).isFalse();
+            // scan again, to get the continuation to the end
+            RecordCursorResult<SizeStatisticsGroupedResults> finalResult = getStoreSizeResult(recordStore, ScanProperties.FORWARD_SCAN, continuationResult.getContinuation().toBytes(), 0);
+            Assertions.assertThat(finalResult.hasNext()).isFalse();
+            Assertions.assertThat(finalResult.getContinuation().isEnd()).isTrue();
+
+            final RecordCursorResult<SizeStatisticsGroupedResults> controlResult = getStoreSizeResult(recordStore);
+
+            Assertions.assertThat(continuationResult.get().getStats()).usingRecursiveComparison().isEqualTo(controlResult.get().getStats());
```

Comment:
Do we need the `usingRecursiveComparison()` thing because `StatisticsResult` doesn't extend `equals`? Should it?

Reply:
Thanks. I'll leave that aside, especially because I don't know what the correct implementation should be due here and for the `GroupedResults`.